### DISCUSSION TITLE
deno quit when exporting oak module

### DISCUSSION THREAD

**Discussion by Srabutdotcom** [denoland/deno#11138] [Q&A]
Dear team,

I use deno version 1.11.2
and have been trying to export oak module but the app quit without any error message.

~~~js
import { Application } from 'https://deno.land/x/oak/mod.ts';
const app = new Application();
export {app}
~~~

and then when i call deno run, it is just quit.....

What is wrong?

**Comment by satyarohith** [comment]
Can you share the code where you're importing the `app`?

**Comment by Srabutdotcom** [comment]
> Can you share the code where you're importing the `app`?

~~~js
import {app} from '../start.js'
await app.listen({ 
      hostname: localhost, 
      port: 80})
~~~

even before reaching this line, deno quit.

**Reply by satyarohith** [reply]
I get the following error with your code. This is expected as there's no middleware to process the request in your code. Are you getting the same error? If not, please provide a link to the complete code and reproduction steps.

```
error: Uncaught TypeError: There is no middleware to process requests.
      throw new TypeError("There is no middleware to process requests.");
            ^
    at Application.listen (https://deno.land/x/oak@v7.6.3/application.ts:471:13)
```

**Reply by Srabutdotcom** [reply]
On my side, there no error pop up like you got.
before this, the app works just fine,.
before this strange behaviour, my environment are:

- windows 10,
- deno 1.10.1 (if i am not mistaken)
- oak v.7.50
- deno installed using scoop

the strage behaviour arises when i want to upgrade deno into the latest version but failed with error "access denied".
And after that i re-installed deno using below code i/o scoop:
~~~js
iwr https://deno.land/x/install/install.ps1 -useb | iex
~~~

deno installed into the latest version into different folder, and i remove scoop path env, and add deno bin into new env path, in order deno run using the latest version under the new folder.

and to investigate this strange behaviour, i create reproduce new simple oak app with the following structure code.

code in **start.js file**
~~~js
import { Application } from 'https://deno.land/x/oak/mod.ts';
const app = new Application();
export {app}
~~~

code in **www**
~~~js
import {app} from '../start.js'
app.use((ctx) => {
   ctx.response.body = "Hello World";
});
await app.listen({port: 80})
~~~

and running the server using the following code:
~~~js
deno run start.js
~~~

and after that deno server is quit, back to command prompt.

I think, it is about my system configuration in windows, but i don't know how to resolve this.
when i inspect using chrome, i found some error ""Failed to construct 'Event'" in deno:extensions/webidl/00_webidl.js under function requiredArguments(length, required, opts = {}) (line no. 571). and i found file:///D:/deno%20apps/experiment/oak/aicone/__anonymous__ add automatically by deno i think. And then the code running into deno:extensions/timers/01_timers.js (line no. 320) looping in this "function handleTimerMacrotask()" without ending. And if i push "resume script execution" deno hang (looping without ending). And if i quit the inspect deno quit back to command prompt.

Any one have solution?

**Reply by Srabutdotcom** [reply]
Sorry,

the command order are wrong, i forgot that deno cli should be targeting to file where there is a method (end point) calling for server i.e listen to port i/o like this...

~~~js
deno run www
~~~

however, i think deno should have a method to detect such human error, i.e. the error code triggering from this mistake, no method called inside the app instead of looping without ending.?

i will keep this discussion open, till there are idea for such preventif solution.

thanks

**Reply by kitsonk** [reply]
How would Deno know it is human error reliably?

**Reply by Srabutdotcom** [reply]
i don't have any idea about this. Maybe like compiling process, checking the code in the main file or method prior running into the production or something like that. Checking process if the main file are containing listening into the port, otherwise throw error message that the main file doesn't have any task to do.

**Comment by crowlKats** [comment]
One thing thats off is the oak import: both single and double ticks are being used in the string for the specifier

**Reply by Srabutdotcom** [reply]
This is not the root cause, this is my mistake putting the code here.
